aten/src/TH/THAllocator.h

```diff
@@ -11,18 +11,19 @@
 #define TH_ALLOCATOR_MAPPED_FROMFD 32
 #define TH_ALLOCATOR_MAPPED_UNLINK 64
 
-/* Custom allocator
- */
-typedef struct THAllocator {
-  void* (*malloc)(void*, ptrdiff_t);
-  void* (*realloc)(void*, void*, ptrdiff_t);
-  void (*free)(void*, void*);
-} THAllocator;
+#ifdef __cplusplus
+#include <ATen/Allocator.h>
+using THAllocator = at::Allocator;
```

Comment:
According to CI, it does work :) (And yes, we need it to be available from C, because a lot of C function headers refer to it).

Actually, I am kind of thinking that we should use `#ifdef __cplusplus` more liberally in the TH headers, as an alternative way of C++ifying headers instead of the `hpp` convention.

Reply:
Oh, I meant does anything from outside of torch (e.g. the CI, internal code) actually use these functions (so they would care if the function decls moved to C++).